src/ert/_c_wrappers/enkf/ensemble_config.py

```diff
@@ -244,13 +244,24 @@ def __init__(
                 self.addNode(node)
 
         for gen_kw in gen_kw_list:
-            gen_kw_node = self.gen_kw_node(gen_kw, tag_format)
             self._create_node_metainfo(gen_kw, 4, EnkfVarType.PARAMETER)
-            self._config_node_meta[gen_kw[0]].output_file = gen_kw[2]
-            init_file = self._config_node_meta[gen_kw[0]].init_file
+            gen_kw_key = gen_kw[0]
+            self._config_node_meta[gen_kw_key].output_file = gen_kw[2]
+            init_file = self._config_node_meta[gen_kw_key].init_file
             if init_file:
-                self._config_node_meta[gen_kw[0]].init_file = os.path.abspath(init_file)
-            self.addNode(gen_kw_node)
+                self._config_node_meta[gen_kw_key].init_file = os.path.abspath(
+                    init_file
+                )
+
+            kw_node = GenKwConfig(
+                key=gen_kw_key,
+                template_file=_get_abs_path(gen_kw[1]),
+                parameter_file=_get_abs_path(gen_kw[3]),
+                tag_fmt=tag_format,
+            )
+
+            self._check_config_node(kw_node)
+            self.add_gen_kw_node(kw_node)
 
         for surface in surface_list:
             surface_node = self.get_surface_node(surface)
```

Comment:
`get_surface_node(..)` returns a `SurfaceConfig` and `self.addNode(..)` accepts a `EnkfConfigNode` | `Field` | `GenKwConfig` .

Reply:
https://github.com/equinor/ert/issues/5309